org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/Conversions.java

```diff
@@ -42,16 +44,142 @@ private Conversions() {
 	}
 
 	/**
-	 * Wraps {@code object} in a list if and only if {@code object} is an array. Works for primitive and
-	 * object-component types.
+	 * Wraps {@code object} in a list if and only if {@code object} is an array.
+	 * Works for primitive and object-component types.
 	 * 
-	 * @param object
-	 *            the object to be wrapped. May be <code>null</code>.
-	 * @return a list if the given object was an array. Otherwise the unmodified given object. May return
-	 *         <code>null</code> if the object was <code>null</code>.
+	 * @param object the object to be wrapped. May be <code>null</code>.
+	 * @return a list if the given object was an array. Otherwise the unmodified
+	 * given object. May return <code>null</code> if the object was
+	 * <code>null</code>.
 	 */
 	@Pure
-	public static Object doWrapArray(Object object) {
+	public static <T> Iterable<T> doWrapArray(T[] object) {
+		return WrappedArray.create(object);
+	}
+
+	/**
+	 * Wraps {@code object} in a list if and only if {@code object} is an array.
+	 * Works for primitive and object-component types.
+	 * 
+	 * @param object the object to be wrapped. May be <code>null</code>.
+	 * @return a list if the given object was an array. Otherwise the unmodified
+	 * given object. May return <code>null</code> if the object was
+	 * <code>null</code>.
+	 */
+	@Pure
+	public static WrappedBooleanArray doWrapArray(boolean[] object) {
```

Comment:
I don't want to have the unfortunately publicly declared types WrappedXXXArray on the signatures of the methods here. What problem to you actually want to solve by this change?